ros2/src/geoscenario_server/geoscenario_server/geoscenario_server.py

```diff
@@ -0,0 +1,258 @@
+
+from rclpy.exceptions import ParameterNotDeclaredException
+from rcl_interfaces.msg import ParameterType, ParameterDescriptor
+
+from geoscenarioserver.GSServerBase import GSServerBase
+from geoscenarioserver.SimTraffic import SimTraffic
+from geoscenarioserver.shm.SimSharedMemoryClient import SimSharedMemoryClient
+from geoscenarioserver.dash.Dashboard import Dashboard, get_screen_parameters
+
+from geoscenario_msgs.msg import Tick, Pedestrian, Vehicle
+from geographic_msgs.msg import GeoPoint
+from lanelet2.io import Origin
+from lanelet2.projection import LocalCartesianProjector
+from lanelet2.core import BasicPoint3d, GPSPoint
+
+@dataclass
+class Parameter:
+    name:str = ""
+    type:ParameterType = ParameterType.PARAMETER_NOT_SET
+    description:str = ""
+    default_value: Any = None
+
+class GSServer(Node, GSServerBase):
+    def __init__(self):
+        Node.__init__(self, 'geoscenario_server')
+        GSServerBase.__init__(self)
+
+        parameters = [
+            Parameter(name='scenario_files', type=ParameterType.PARAMETER_STRING_ARRAY, description='GeoScenario file paths', default_value=['']),
+            Parameter(name='no_dashboard', type=ParameterType.PARAMETER_BOOL, description='run without the dashboard', default_value=False),
+            Parameter(name='map_path', type=ParameterType.PARAMETER_STRING, description='Set the prefix to append to the value of the attribute `globalconfig->lanelet`', default_value=""),
+            Parameter(name='btree_locations', type=ParameterType.PARAMETER_STRING, description='Add higher priority locations to search for btrees by agent btypes', default_value=""),
+            Parameter(name='dashboard_position', type=ParameterType.PARAMETER_DOUBLE_ARRAY, description='Set the position of the dashboard window (x y width height)', default_value=[0.0]),
+            Parameter(name='wgs84', type=ParameterType.PARAMETER_BOOL, description='Use WGS84+origin(0,0,0) coordinates instead of local+origin=(lat,lon,alt) coordinates', default_value=False)
+        ]
+
+        for param in parameters:
+            param_descriptor = ParameterDescriptor(type=param.type, description=param.description)
+            self.declare_parameter(param.name, param.default_value, param_descriptor)
+
+        self.previous_tick_count = -1
+        self.tick_count = 0
+        self.simulation_time = 0.0
+
+        self.wgs84 = self.get_parameter('wgs84').get_parameter_value().bool_value
+        self.sim_config.show_dashboard = not self.get_parameter('no_dashboard').get_parameter_value().bool_value
+
+        btree_locations_param = self.get_parameter('btree_locations').get_parameter_value().string_value
+        btree_locations = self.parse_btree_paths(btree_locations_param)
+        
+        self.traffic = SimTraffic(self.lanelet_map, self.sim_config)
+
+        # for now since the SimTraffic reads from what the client writes, we need to still involve the shm
+        self.sim_client_shm = SimSharedMemoryClient()
```

Comment:
can we try to eliminate this? We can modify simtraffic to skip reading/writing to shm when this is set to `None`.